src/mcts/search.cc

```diff
@@ -1395,39 +1401,8 @@ void SearchWorker::DoBackupUpdateSingleNode(
     // Nothing left to do without ancestors to update.
     if (!p) break;
 
-    // Convert parents to terminals except the root or those already converted.
-    can_convert = can_convert && p != search_->root_node_ && !p->IsTerminal();
-
-    // A non-winning terminal move needs all other moves to be similar.
-    auto all_losing = true;
-    auto found_tb = n->IsTbTerminal();
-    float losing_m = 0.0f;
-    if (can_convert && v <= 0.0f) {
-      for (const auto& edge : p->Edges()) {
-        const auto WL = edge.GetWL();
-        can_convert = can_convert && edge.IsTerminal() && WL <= 0.0f;
-        if (!can_convert) break;
-        all_losing = all_losing && WL < 0.0f;
-        found_tb = found_tb || edge.IsTbTerminal();
-        losing_m = std::max(losing_m, edge.GetM(0.0f));
-      }
-    }
-
-    // Convert the parent to a terminal loss if at least one move is winning or
-    // to a terminal win if all moves are losing; otherwise there's a mix of
-    // draws and losing, so at best it's a draw.
-    if (can_convert) {
-      // Doesn't give the correct distance to mate because siblings are not
-      // considered but more accurate than doing nothing. This shouldn't
-      // underestimate the distance to mate since at worst we miss shorter
-      // moves.
-      float terminal_m = std::max(losing_m, m) + 1.0f;
-      p->MakeTerminal(
-          v > 0.0f ? GameResult::BLACK_WON
-                   : all_losing ? GameResult::WHITE_WON : GameResult::DRAW,
-          terminal_m,
-          found_tb ? Node::Terminal::Tablebase : Node::Terminal::Terminal);
-    }
+    // A new terminal among its siblings might require converting the parent.
+    can_convert = MaybeConvertParent(p, can_convert, m);
```

Comment:
I think this would be clearer as
```c++
if (can_convert) {
  can_convert = MaybeConvertParent(p, m);
}
```
or similar to the old code
```c++
can_convert = can_convert && MaybeConvertParent(p, m);
```

Reply:
Switched back more to like the old code chaining `&&`s